mne/connectivity/cfc.py

```diff
@@ -0,0 +1,393 @@
+    evoked_raw = np.zeros(len(times))
+    evoked_spec_amplitude = np.zeros((n_freqs, len(times)), dtype=np.float)
+    for ind in trigger_inds:
+        evoked_raw += data[0, ind - n_times_min: ind + n_times_max]
+        this_spec = raw_spec_amplitude[:, ind - n_times_min: ind + n_times_max]
+        evoked_spec_amplitude += this_spec
+    evoked_raw /= len(trigger_inds)
+    evoked_spec_amplitude /= len(trigger_inds)
+
+    return (times, raw_spec_amplitude, evoked_spec_amplitude,
+            evoked_raw, trigger_inds)
+
+
+def phase_binned_amplitude(data_amp, sfreq, fmin_phase, fmax_phase,
+                           fmin_amp, fmax_amp, data_phase=None,
+                           n_bins=18, method='fft', surrogates=False,
+                           n_jobs=1):
+    """ Compute normalized amplitude histograms binned by low-freq phase.
+
+    Parameters
+    ----------
+    data_amp : ndarray (n_epochs, n_times_epoch)
+        Raw signal time series to be used for amplitude component of PAC.
+    sfreq : float
+        Sampling frequency in Hz.
+    fmin_phase : float
+        Lower edge of phase modulating frequency.
+    fmax_phase : float
+        Upper edge of phase modulating frequency.
+    fmin_amp : float
+        Lower edge of amplitude modulated frequency.
+    fmax_amp : float
+        Upper edge of amplitude modulated frequency.
+    data_phase : ndarray (n_epochs, n_times_epoch) | None
+        Raw signal time series for phase calculation in PAC. If None,
+        will be the same as data_amp. Must be the same shape as data_amp.
+    n_bins : int
+        Number of phase bins. One full cycle is divided into equally-spaced
+        bins. (20 degrees * 18 = 360)
+    method : str
+        Filter method used. Default is 'fft'.
+    surrogates : bool
+        Return surrogates values for data specific to PAC,
+        the phase series and amplitude envelope are shuffled for every trial.
+        Default value is False.
+    n_jobs : int
+        Number of jobs used for parallelization. Default 1.
+
+    Returns
+    -------
+    normalized_amplitudes : ndarray, shape (n_epochs, n_bins)
+        The normalized amplitude values for the given frequency amplitude
+        across each epoch/bin.
+    phase_bins : array, shape (n_bins + 1,)
+        Binned phase points. Used for plotting.
+
+    References
+    ----------
+    Measuring phase-amplitude coupling between neuronal oscillations
+    of different frequencies.
+    Tort ABL, Komorowski R, Eichenbaum H, Kopell N., (J Neurophysiol. 2010)
+    """
+    from scipy.signal import hilbert
+
+    if data_phase is None:
+        data_phase = data_amp
+    if data_amp.shape != data_phase.shape:
+        raise AssertionError('Signals must be the same length and n_epochs.')
+
+    # obtained number of trials
+    n_epochs, n_times = data_amp.shape
+
+    # reshape epochs into single time series for filtering
+    filter_data_amp = data_amp.reshape((1, n_times * n_epochs))[0]
```

Comment:
yep. Changed the reshapes to either use `ravel`, or `np.newaxis` depending on what they were trying to do.